File: codewars/8kyu/alternating-case/dduck-alternating-case.py
# https://www.codewars.com/kata/alternating-case-%3C-equals-%3E-alternating-case

def to_alternating_case(string):
    return "".join([(lambda c: c.lower() if c.isupper() else c.upper())(char) for char in string])


print('a'.islower())
print(to_alternating_case('aBcDeF'))

File: codewars/8kyu/alternating-case/dduck-alternating-case.js
// https://www.codewars.com/kata/alternating-case-%3C-equals-%3E-alternating-case

String.prototype.toAlternatingCase = function () {
  return this.split('')
    .map((char) => (char === char.toUpperCase() ? char.toLowerCase() : char.toUpperCase()))
    .join('');
};

console.log('aBcDeF'.toAlternatingCase());
//?
console.log('aBcDeF'.toUpperCase());
//?
